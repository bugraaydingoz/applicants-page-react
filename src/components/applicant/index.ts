export { Applicant } from './Applicant'
export { ApplicantList } from './ApplicantList'
