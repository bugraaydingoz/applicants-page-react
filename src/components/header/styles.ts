import styled from 'styled-components'

export const HeaderContainer = styled.header`
  display: flex;
  justify-content: space-between;
  margin: var(--space-sm) var(--space-md);
`
