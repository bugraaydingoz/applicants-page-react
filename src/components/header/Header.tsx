import React from 'react'

// Hooks
import { useViewport } from '../../hooks/useViewport'
import { useTranslation } from 'react-i18next'

// Components
import { ReactComponent as MenuIcon } from '../shared/icons/menu.svg'
import { ReactComponent as Logo } from '../shared/icons/logo.svg'
import { ReactComponent as MessageActiveIcon } from '../shared/icons/message-active.svg'
import { ReactComponent as ContactIcon } from '../shared/icons/contact.svg'
import { ReactComponent as UserIcon } from '../shared/icons/user.svg'
import { ReactComponent as LogoutIcon } from '../shared/icons/logout.svg'

// Styles
import { HeaderContainer, ActionsContainer, ActionContainer } from './styles'
import { H5 } from '../shared/typography'

export function Header() {
  const { width } = useViewport()
  const isMobile = width <= 768
  return (
    <HeaderContainer>
      {isMobile ? <HeaderMobile /> : <HeaderDesktop />}
    </HeaderContainer>
  )
}

function HeaderMobile() {
  return (
    <>
      <MenuIcon />
      <Logo style={{ width: '100' }} />
      <MessageActiveIcon />
    </>
  )
}

function HeaderDesktop() {
  const { t } = useTranslation()
  return (
    <>
      {/* Left buttons */}
      <ActionsContainer>
        <ActionContainer>
          <MenuIcon />
        </ActionContainer>
        <Logo />
      </ActionsContainer>

      {/* Right buttons */}
      <ActionsContainer>
        <ActionContainer style={{ width: 'auto' }}>
          <ContactIcon style={{ marginRight: 'var(--space-xs' }} />
          <H5>{t('header.contactSupport')}</H5>
        </ActionContainer>
        <ActionContainer>
          <MessageActiveIcon />
        </ActionContainer>
        <ActionContainer>
          <UserIcon />
        </ActionContainer>
        <ActionContainer>
          <LogoutIcon />
        </ActionContainer>
      </ActionsContainer>
    </>
  )
}
