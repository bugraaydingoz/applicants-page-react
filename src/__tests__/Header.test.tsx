import React from 'react'
import renderer from 'react-test-renderer'
import { Header } from '../components/header'

test('renders correctly', () => {
  const tree = renderer.create(<Header />).toJSON()
  expect(tree).toMatchSnapshot()
})
