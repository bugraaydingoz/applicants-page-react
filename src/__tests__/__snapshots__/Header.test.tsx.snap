// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="sc-AxjAm biOILs"
>
  <div
    className="sc-AxirZ hlnYRa"
  >
    <div
      className="sc-AxiKw bWcrCB"
    >
      <svg>
        menu.svg
      </svg>
    </div>
    <svg>
      logo.svg
    </svg>
  </div>
  <div
    className="sc-AxirZ hlnYRa"
  >
    <div
      className="sc-AxiKw bWcrCB"
      style={
        Object {
          "width": "auto",
        }
      }
    >
      <svg
        style={
          Object {
            "marginRight": "var(--space-xs",
          }
        }
      >
        contact.svg
      </svg>
      <h5
        className="sc-AxhCb evMNyv"
      >
        header.contactSupport
      </h5>
    </div>
    <div
      className="sc-AxiKw bWcrCB"
    >
      <svg>
        message-active.svg
      </svg>
    </div>
    <div
      className="sc-AxiKw bWcrCB"
    >
      <svg>
        user.svg
      </svg>
    </div>
    <div
      className="sc-AxiKw bWcrCB"
    >
      <svg>
        logout.svg
      </svg>
    </div>
  </div>
</header>
`;
